build-logic/api-extractor/src/test/groovy/org/gradle/internal/tools/normalization/java/ApiClassExtractorTest.groovy

```diff
@@ -274,10 +271,7 @@ class ApiClassExtractorTest extends ApiClassExtractorTestSupport {
         'boolean' | 'true'         | false
     }
 
-    void "target binary compatibility is maintained"() {
-        Assume.assumeFalse(target == "1.6" && !canTargetJava6())
-        Assume.assumeFalse(target == "1.7" && !canTargetJava7())
-
+    void "target binary compatibility is maintained for Java version #target"() {
```

Comment:
💭 Simplified this by testing with Java versions that are actually supported on all our agents.